weblab/entities/views.py

```diff
@@ -794,6 +795,49 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class TransferView(LoginRequiredMixin, UserPassesTestMixin,
+                   FormMixin, EntityTypeMixin, DetailView):
+    template_name = 'entities/entity_transfer_ownership.html'
+    context_object_name = 'entity'
+    form_class = OwnershipTransferForm
+
+    def _get_object(self):
+        if not hasattr(self, 'object'):
+            self.object = self.get_object()
+        return self.object
+
+    def test_func(self):
+        return self._get_object().is_managed_by(self.request.user)
+
+    def post(self, request, *args, **kwargs):
+        """Check the form and transfer ownership of the entity.
+
+        Called by Django when a form is submitted.
+        """
+        form = self.get_form()
+
+        if form.is_valid():
+            user = form.cleaned_data['user']
+            entity = self.get_object()
+            if self.model.objects.filter(name=entity.name, author=user).exists():
+                form.add_error(None, "User already has a %s called %s" % (self.model.display_type, entity.name))
+                return self.form_invalid(form)
+
+            old_path = entity.repo_abs_path
+            entity.author = user
+            entity.save()
+            user.get_storage_dir('repo').mkdir(exist_ok=True, parents=True)
+            new_path = entity.repo_abs_path
```

Comment:
Looking at the latest failure, this might need to change: I think when a new spec is created it automatically creates all parent folders, whereas this misses the last 'fittingspec' segment. Instead we could do:
```suggestion
            new_path = entity.repo_abs_path
            new_path.parent.mkdir(exist_ok=True, parents=True)
```
which makes sure all levels above where we're moving to are created.